app/src/main/java/com/willowtree/vocable/splash/SplashViewModel.kt

```diff
@@ -21,7 +26,51 @@ class SplashViewModel : BaseViewModel() {
     private fun populateDatabase() {
         backgroundScope.launch {
             presetsRepository.populateDatabase()
+
+            moveMySayings()
+
             liveExitSplash.postValue(true)
         }
     }
-}
\ No newline at end of file
+
+    private suspend fun moveMySayings() {
+        // if the user has My Sayings phrases, we need to migrate them to a custom category
+        if (presetsRepository.getPhrasesForCategory(PresetCategories.USER_FAVORITES.id).isNotEmpty()) {
+            val allCategories = presetsRepository.getAllCategories()
+
+            // Get the index of the first hidden category to find the sort order of new category
+            var firstHiddenIndex = allCategories.indexOfFirst { it.hidden }
+            if (firstHiddenIndex == -1) {
+                firstHiddenIndex = allCategories.size
+            }
+
+            // Create a new custom category called "My Sayings" that is user-generated
+            val newCategoryId = UUID.randomUUID().toString()
+            val newCategory = Category(
+                newCategoryId,
+                System.currentTimeMillis(),
+                true,
+                null,
+                mapOf(Pair(Locale.getDefault().toString(), "My Sayings")),
+                false,
+                firstHiddenIndex
+            )
+
+            presetsRepository.addCategory(newCategory)
+
+            // Get the phrases from the old My Sayings category and add cross refs with the new category
+            val mySayingsPhrases = presetsRepository.getPhrasesForCategory(PresetCategories.USER_FAVORITES.id)
+            mySayingsPhrases.forEach {
+                presetsRepository.addCrossRef(CategoryPhraseCrossRef(newCategoryId, it.phraseId))
+            }
+
+            // Get the old My Sayings category and delete it
+            val mySayingsCategory = presetsRepository.getCategoryById(PresetCategories.USER_FAVORITES.id)
```

Comment:
Should we also delete the old category if the user didn't have any saved phrases? That way it's not hanging around in memory.

Reply:
Yeah, that's what the next line is doing, right?  It gets the old My Sayings category on line 68, then deletes it on line 69

Reply:
jk I see what you're saying, you're right